File: Cards/Banner.ts
/** @noSelfInFile */

import { Ranks, Suits } from "../Enums"
import Card from "./Card"
import GameManager from "../GameManager"

export default class Banner extends Card {
    constructor(gameManager: GameManager, suit: Suits) {
        super(gameManager, suit, Ranks.BANNER, 6, 3, "Banner")
    }
}

File: Cards/Banner.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__ClassExtends = ____lualib.__TS__ClassExtends
local ____exports = {}
local ____Enums = require("Enums")
local Ranks = ____Enums.Ranks
local ____Card = require("Cards.Card")
local Card = ____Card.default
____exports.default = __TS__Class()
local Banner = ____exports.default
Banner.name = "Banner"
__TS__ClassExtends(Banner, Card)
function Banner.prototype.____constructor(self, gameManager, suit)
    Card.prototype.____constructor(
        self,
        gameManager,
        suit,
        Ranks.BANNER,
        6,
        3,
        "Banner"
    )
end
return ____exports
